addon/instance-initializers/prefetch.js

```diff
@@ -52,16 +52,48 @@ export function initialize(instance) {
             if (handler.isDestroying === true) {
               return;
             }
-            handler._prefetched = handlerInfo.runSharedModelHook(transition, 'prefetch', [fullParams]);
+            handler._prefetched = runHook('prefetch', handlerInfo, transition, [fullParams]);
           })
         ));
       } else {
-        handlerInfo.handler._prefetched = handlerInfo.runSharedModelHook(transition, 'prefetch', [fullParams]);
+        handlerInfo.handler._prefetched = runHook('prefetch', handlerInfo, transition, [fullParams]);
       }
     });
   });
 }
 
+function runHook(hookName, handlerInfo, transition, args) {
+  /*
+    `runSharedModelHook` was deleted as part of an internal cleanup
+    and is now moved to a function much like this one. This detects
+    if the `runSharedModelHook` exists or not.
```

Comment:
can you add a comment which explains versions is this for, and when can it be removed.